File: fral/examples/basic.cpp
#include "../engine/engine.h"
#include <iostream>

int main() {

    //Process 1
    auto ralA = fral::FRAL("test.bin", 1000, 10);
    const char *TEST_STR = "TEST";

    auto blob = (char *) ralA.allocate(strlen(TEST_STR));
    strcpy(blob, TEST_STR);
    printf("%s\n", blob);

    ralA.append(blob);


    //Process 2
    auto ralB = fral::FRAL("test.bin");

    auto blob2 = (char *) ralB.load(0);
    printf("%s\n", blob2);
    blob2[0] = 'D';

    auto blob3 = (char *) ralB.load(0);
    printf("%s\n", blob3);
}

File: _fral/examples/basic.py
from fral.python.fral.fral import FRAL


def main():

    # Process 1
    ral_A = FRAL("test.bin", 1000, 100)
    test_bytes = "TEST".encode()

    test_blob = ral_A.allocate(len(test_bytes))
    test_blob[:len(test_bytes)] = test_bytes
    print(bytes(test_blob).decode())
    ral_A.append(test_blob)

    # Process 2
    ral_B = FRAL("test.bin")

    test_blob2 = ral_B.read(0)
    print(bytes(test_blob2).decode())
    test_blob2[0:1] = 'D'.encode()

    test_blob3 = ral_B.read(0)
    print(bytes(test_blob3).decode())


if __name__ == '__main__':
    main()




